app/screens/backup/BackupRoot.js

```diff
@@ -0,0 +1,193 @@
+  width: 70px;
+  height: 15px;
+`;
+
+const MiddleSection = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 502px;
+  height: 100%;
+  padding: 25px 15px 15px 15px;
+  background-color: ${smColors.black02Alpha};
+`;
+
+const MiddleSectionRow = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const BottomSection = styled(MiddleSectionRow)`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  justify-content: flex-end;
+`;
+
+const BottomRow = styled(MiddleSectionRow)`
+  justify-content: space-between;
+`;
+
+type Props = {
+  wallet: WalletMeta,
+  history: RouterHistory
+};
+
+class BackupRoot extends Component<Props> {
+  render() {
+    return (
+      <Wrapper>
+        <LeftSection>
+          <WrapperWith2SideBars width={300} height={480} header="BACKUP">
+            <BoldText>How would you like to backup your wallet?</BoldText>
+            <Text>Your wallet is encrypted with your password on your computer, but we recommend you backup your wallet for additional security.</Text>
+          </WrapperWith2SideBars>
+        </LeftSection>
+        <RightSection>
+          <TopLeftCorner src={topLeftCorner} />
+          <TopRightCorner src={topRightCorner} />
+          <BottomLeftCorner src={bottomLeftCorner} />
+          <BottomRightCorner src={bottomRightCorner} />
+          <HorizontalBar src={smallHorizontalSideBar} />
+          <MiddleSection>
+            <MiddleSectionRow>
+              <SmallText style={{ marginRight: 24, width: 224 }}>Basic Security</SmallText>
+              <SmallText style={{ width: 224 }}>
+                Advanced Security <GreenText>(Recommended)</GreenText>
+              </SmallText>
+            </MiddleSectionRow>
+            <MiddleSectionRow>
+              <Button onClick={this.backupWallet} text="File Backup" isPrimary={false} width={224} style={{ marginRight: 24 }} />
+              <Button onClick={this.navigateTo12WordsBackup} text="12 Words Backup" isPrimary={false} width={224} />
+            </MiddleSectionRow>
+            <BottomSection>
+              <BottomRow>
+                <Link onClick={this.openBackupGuide} text="BACKUP GUIDE" style={{ paddingTop: 26 }} />
+              </BottomRow>
+            </BottomSection>
+          </MiddleSection>
+        </RightSection>
+      </Wrapper>
+    );
+  }
+
+  navigateTo12WordsBackup = () => {
+    const { history } = this.props;
+    history.push('/main/backup/twelve-words-backup');
+  };
+
+  backupWallet = async () => {
+    const { wallet, history } = this.props;
+    try {
+      const { meta, accounts, mnemonic, transactions, contacts, fileKey } = wallet;
+      const encryptedAccountsData = fileEncryptionService.encryptData({ data: JSON.stringify({ mnemonic, accounts }), key: fileKey });
+      const encryptedWallet = { meta, crypto: { cipher: 'AES-128-CTR', cipherText: encryptedAccountsData }, transactions, contacts };
+      const fileName = `Wallet_Backup_${Date().toString()}.json`;
```

Comment:
have you checked how the file name looks like eventually? Does "Wallet_Backup_Thu Aug 15 2019 18/00/16 GMT+0300 (Israel Daylight Time)" looks fine for you?

Reply:
change to ISO string YYYY-MM-DDTHH:MM:SS.mmmZ